.github/workflows/update_wrgl_urls.yml

```diff
@@ -0,0 +1,172 @@
+name: Update WRGL URLs
+
+on:
+  workflow_run:
+    workflows: ["Upload CSV to Google Cloud"]
+    types:
+      - completed
+
+jobs:
+  update-urls:
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      
+      # Set up Cloud SQL Auth proxy
+      - id: 'auth'
+        uses: 'google-github-actions/auth@v1'
+        with:
+          credentials_json: '${{ secrets.GCP_SA_KEY }}'
+          
+      - name: 'Set up Cloud SQL Proxy'
+        uses: 'google-github-actions/setup-cloud-sql-proxy@v1'
+        with:
+          instance: '${{ secrets.GCP_PROJECT_ID }}:us-central1:${{ secrets.CLOUD_SQL_DATABASE }}'
+          port: 5432
+
+      - name: Download and verify csv_urls.json
+        run: |
+          FOLDER_NAME="data-${{ github.event.workflow_run.id }}-${{ github.event.workflow_run.run_attempt }}"
+          echo "Checking for csv_urls.json in bucket..."
+          if ! gsutil -q stat "gs://${{ vars.DATA_BUCKET }}/$FOLDER_NAME/csv_urls.json"; then
+            echo "Error: csv_urls.json not found in cloud storage"
+            exit 1
+          fi
+          echo "Downloading csv_urls.json..."
+          gsutil cp "gs://${{ vars.DATA_BUCKET }}/$FOLDER_NAME/csv_urls.json" ./csv_urls.json
+          echo "Verifying JSON format..."
+          if ! jq empty csv_urls.json; then
+            echo "Error: Invalid JSON format in csv_urls.json"
+            exit 1
+          fi
+          echo "Verifying JSON content..."
+          URL_COUNT=$(jq 'length' csv_urls.json)
+          echo "Found $URL_COUNT URL entries in csv_urls.json"
+          if [ "$URL_COUNT" -eq "0" ]; then
+            echo "Error: No URLs found in csv_urls.json"
+            exit 1
+          fi
+
+      - name: Update Database
+        env:
+          POSTGRES_DB: ${{ secrets.POSTGRES_DB }}
+          POSTGRES_USER: ${{ secrets.POSTGRES_USER }}
+          POSTGRES_PASSWORD: ${{ secrets.POSTGRES_PASSWORD }}
+        run: |
+          sudo apt-get update && sudo apt-get install -y python3-pip
+          pip install psycopg2-binary
+          cat << 'EOF' > update_wrgl.py
+          import psycopg2
+          import os
+          import json
+          import re
+
+          def get_slug_from_filename(filename):
+              match = re.match(r'[a-z]+_(.+)\.csv', filename)
+              if match:
+                  return match.group(1).replace('_', '-')
+              return None
+
+          def get_file_type(filename):
+              match = re.match(r'([a-z]+)_', filename)
+              if match:
+                  return match.group(1)
+              return None
+
+          def update_wrgl_urls():
+              success_count = 0
+              error_count = 0
+              skipped_count = 0
+              print("Starting database update process...")
+              try:
```

Comment:
i struggled a bit following the triply-nested try/except blocks. at least the first one seems like it could be pulled out -- if we can't establish a db connection then nothing else is going to happen anyways.

Reply:
(by "pulled out") i just mean into its own block

Reply:
yea, you're right. cleaned it up a bit here: https://github.com/ipno-llead/processing/pull/575/commits/e88fff368798bf154c0cf987ca9579d25bc69561